core/src/Cuda/Kokkos_Cuda_Locks.cpp

```diff
@@ -59,7 +59,7 @@ void initialize_host_cuda_lock_arrays() {
                  sizeof(int) * (CUDA_SPACE_ATOMIC_MASK + 1)));
   Impl::cuda_device_synchronize(
       "Kokkos::Impl::initialize_host_cuda_lock_arrays: Pre Init Lock Arrays");
-  g_host_cuda_lock_arrays.n = Cuda::concurrency();
+  g_host_cuda_lock_arrays.n = CudaInternal::concurrency();
```

Comment:
It looks like there are mixed usages between `CudaInternal::concurrency()` and `Cuda().concurrency()`. Is there a reason to prefer `CudaInternal::` over `Cuda()...` here?

Reply:
Yes.  Chicken or the egg dilemma:  `initialize_host_cuda_lock_arrays` is called as you initialize the backend so we really can't construct an execution space here